packages/nodes-base/credentials/MicrosoftOAuth2Api.credentials.ts

```diff
@@ -33,6 +33,16 @@ export class MicrosoftOAuth2Api implements ICredentialType {
 			type: 'string',
 			default: 'https://login.microsoftonline.com/common/oauth2/v2.0/token',
 		},
+		{
+			displayName: 'Microsoft Graph API Base URL',
+			name: 'baseUrl',
+			type: 'string',
```

Comment:
This should probably be a dropdown / options field with the options being available so that users can't input the incorrect value.

Does this change also impact oauth or is that always the same url no matter what instance is used?

Reply:
That hardcoded URL is always the same and even when you connect to a non-public registered Azure app it breaks when you try to send an API call, because it always sends to the public tenant URL.

Reply:
I changed it from user input to a dropdown selection for the 4 published graph api endpoints available.  One public and 3 national.  https://learn.microsoft.com/en-us/graph/deployments